employee/employee-cinvoke.c

```diff
@@ -0,0 +1,84 @@
+/* This program  atomically unseal a pair of capabilities
+ * (code and data) by calling `ldpblr`, i.e.
+ * "Load Pair of capabilities and Branch with Link".
+ * See "ARM Architecture Reference Manual Supplement Morello"
+ * (https://developer.arm.com/documentation/ddi0606/latest).
+ */
+
+#include "../include/common.h"
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/sysctl.h>
+
+struct cheri_object
+{
+	void *__capability codecap;
+	void *__capability datacap;
+};
+
+void print_salary(uint8_t salary);
+void invoke(struct cheri_object * pair);
+
+int main()
+{
+
+	// Request special capability `sealcap` from the operating system
+	// in order to use it as key to seal our capabilities
+	void *__capability sealcap;
+	size_t sealcap_size = sizeof(sealcap);
+	if (sysctlbyname("security.cheri.sealcap", &sealcap, &sealcap_size, NULL, 0) < 0)
+	{
+		error("Fatal error. Cannot get `securiy.cheri.sealcap`.");
+		exit(1);
+	}
```

Comment:
At least on Morello, this isn't actually required, because there's a `SEAL (immediate)` instruction that can seal one of the special CInvoke types without a `sealcap`.

Reply:
aha, 4.4.122 on the Extended ISA, amazing! I'll fix this.

Reply:
Tried without sealing and it calls the target and hangs, in 5faa64e.

Reply:
After using `seal (immediate`, it does not seem to seal anything (the assertions fail). It jumps to the right location and it hangs after printing the `salary`. In f7d6b23.